js/piwik.js

```diff
@@ -7706,6 +7719,54 @@ if (typeof window.Piwik !== 'object') {
         // initialize the Piwik singleton
         addEventListener(windowAlias, 'beforeunload', beforeUnloadHandler, false);
 
+        window.addEventListener('message', function(e) {
+            var tracker = Piwik.getTracker();
+
+            var originHost = getHostName(e.origin);
+            var matomoHost = getHostName(tracker.getPiwikUrl());
```

Comment:
Would maybe need to adjust `getHostName` to check if `url` is set... otherwise if the tracking code is embedded on the page, but no async tracker is created (which some users do), then it might not return a string? Not sure though... looking at the code it seems `getPiwikUrl` always returns a string so it is likely fine